cmake/module/TargetDataSources.cmake

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+function(target_data_sources target)
```

Comment:
I think this would make sense as 2 separate functions. Seems a weird mix of args otherwise.
`target_raw_data_source` and `target_json_data_source`?

I think it'd be fine to call both funcs for targets that need both types of files generated.

Reply:
Thanks! [Reworked](https://github.com/bitcoin/bitcoin/pull/30901#issuecomment-2656487136).